cups/file.c

```diff
@@ -335,6 +335,135 @@ _cupsFileCheckFilter(
 #endif /* !_WIN32 */
 
 
+/*
+ * '_cupsFileGetConfAndComments()' - Get line and comments from a configuration file.
+ */
+
+char *					/* O  - Line read or @code NULL@ on end of file or error */
+_cupsFileGetConfAndComments(cups_file_t *fp,	/* I  - CUPS file */
+                char        *buf,	/* O  - String buffer */
+		size_t      buflen,	/* I  - Size of string buffer */
+                char        **value,	/* O  - Pointer to value */
+		int         *linenum)	/* IO - Current line number */
+{
+  char	*ptr;				/* Pointer into line */
+
+
+ /*
+  * Range check input...
+  */
+
+  DEBUG_printf(("2cupsFileGetConfAndComments(fp=%p, buf=%p, buflen=" CUPS_LLFMT
+                ", value=%p, linenum=%p)", (void *)fp, (void *)buf, CUPS_LLCAST buflen, (void *)value, (void *)linenum));
+
+  if (!fp || (fp->mode != 'r' && fp->mode != 's') ||
+      !buf || buflen < 2 || !value)
+  {
+    if (value)
+      *value = NULL;
+
+    return (NULL);
+  }
+
+ /*
+  * Read the next line...
+  */
+
+  *value = NULL;
+
+  while (cupsFileGets(fp, buf, buflen))
+  {
+    (*linenum) ++;
+
+    /*
+     * Remove the inline comment...
+     */
+     if ((ptr = strchr(buf, '#')) != NULL)
+     {
+        while(ptr > buf)
+        {
+         // Null-terminate the string after the last non-whitespace, unless the '#' character is escaped by a backslash ('\')
+         if(!_cups_isspace(ptr[-1]) && (ptr == buf || ptr[-1] != '\\'))
+         {
+           *ptr = '\0';
+           break;
+         }
+         ptr--;
+        }
+     }
```

Comment:
Incorrect indentation - every line (except for line with opening multiblock comment ```/*```) is indented by even number of whitespaces - (2 spaces -> 4 spaces -> 6 spaces -> 1 tab -> 1 tab and 2 spaces etc). The initial indentation here is 5, which is odd number.